### PR TITLE
fix: better verbose output

### PR DESCRIPTION
**Pull request by aman29271**
**What kind of change does this PR introduce?**
This PR produces the following changes:

- `Compiler.js` 
Using `--progress` outputs all stages of compilation, percentage included. Previously, Output were overwritten by each other and only the final output got displayed. It also displays all stats information either provided through multiple entry or multiple config.
- `StatsGroup.js`
The information needed by the user to display should be in `outputOptions`. 

**Did you add tests for your changes?**
Yes, 
**If relevant, did you update the documentation?**
No, i am awaiting for feedback.
**Summary**

**Does this PR introduce a breaking change?**
No.

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/utils/Compiler.js [hunk 1]
```diff
@@ -20,8 +22,8 @@ class Compiler {
                 process.stdout.write('\n');
                 const defaultProgressPluginHandler = (percent, msg) => {
                     percent = Math.floor(percent * 100);
-                    process.stdout.clearLine();
-                    process.stdout.cursorTo(0);
+                    process.stdout.write(os.EOL);
+                    readline.cursorTo(process.stdout, 0);
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/utils/Compiler.js:26]: Why you change it?

Comment (aman29271) [reply packages/webpack-cli/lib/utils/Compiler.js:26]: It seems like it was overwriting the msg produced by progressPlugin. Previously, the output produced from plugin look like this.
```bash
(###########) 100% : Complilation completed
```
and after changing it looks more verbose
```bash
(###########) 0% : compiling
 (###########) 10% : building
 (###########) 10% : building
 (###########) 10% : building
 (###########) 10% : building
 (###########) 10% : building
 (###########) 10% : building
 (###########) 10% : building
 (###########) 70% : building
 (###########) 70% : finish module graph
 (###########) 70% : finish module graph
 (###########) 70% : finish module graph
 (###########) 70% : sealing
 (###########) 70% : sealing
 (###########) 72% : basic dependencies optimization
 (###########) 72% : dependencies optimization
 (###########) 72% : dependencies optimization
 (###########) 72% : dependencies optimization
 (###########) 73% : advanced dependencies optimization
 (###########) 73% : after dependencies optimization
 (###########) 71% : chunk graph
 (###########) 71% : after chunk graph
 (###########) 71% : after chunk graph
 (###########) 74% : optimizing
 (###########) 74% : basic module optimization
 (###########) 75% : module optimization
 (###########) 75% : advanced module optimization
 (###########) 76% : after module optimization
 (###########) 76% : basic chunk optimization
 (###########) 76% : basic chunk optimization
 (###########) 76% : basic chunk optimization
 (###########) 76% : basic chunk optimization
 (###########) 76% : basic chunk optimization
 (###########) 77% : chunk optimization
 (###########) 77% : advanced chunk optimization
 (###########) 77% : advanced chunk optimization
 (###########) 77% : advanced chunk optimization
 (###########) 77% : after chunk optimization
 (###########) 78% : module and chunk tree optimization
 (###########) 78% : after module and chunk tree optimization
 (###########) 79% : basic chunk modules optimization
 (###########) 80% : chunk modules optimization
 (###########) 80% : chunk modules optimization
 (###########) 80% : advanced chunk modules optimization
 (###########) 81% : after chunk modules optimization
 (###########) 81% : module reviving
 (###########) 81% : module reviving
 (###########) 82% : module order optimization
 (###########) 82% : module order optimization
 (###########) 82% : advanced module order optimization
 (###########) 83% : before module ids
 (###########) 83% : module ids
 (###########) 84% : module id optimization
 (###########) 84% : module id optimization
 (###########) 85% : chunk reviving
 (###########) 85% : chunk reviving
 (###########) 85% : chunk order optimization
 (###########) 85% : chunk order optimization
 (###########) 86% : before chunk ids
 (###########) 86% : chunk id optimization
 (###########) 86% : chunk id optimization
 (###########) 87% : after chunk id optimization
 (###########) 87% : record modules
 (###########) 87% : record modules
 (###########) 87% : record chunks
 (###########) 87% : record chunks
 (###########) 88% : hashing
 (###########) 88% : after hashing
 (###########) 89% : record hash
 (###########) 89% : module assets processing
 (###########) 90% : chunk assets processing
 (###########) 90% : additional chunk assets processing
 (###########) 91% : recording
 (###########) 92% : additional asset processing
 (###########) 92% : chunk asset optimization
 (###########) 92% : chunk asset optimization
 (###########) 93% : after chunk asset optimization
 (###########) 93% : asset optimization
 (###########) 94% : after asset optimization
 (###########) 94% : after seal
 (###########) 95% : emitting
 (###########) 98% : after emitting
 (###########) 98% : after emitting
 (###########) 100% : Complilation completed
```

Comment (aman29271) [reply packages/webpack-cli/lib/utils/Compiler.js:26]: Also there is some issue in using `process.stdout.clearLine()` in tests. It says that it's not defined or something.

**Comment by aman29271** [comment]
After including `ProgressPlugin` in configuration file. Through debugging I found out  `progerssPluginExists` is still undefined.
```javascript
const { ProgressPlugin } = webpack;
        let progressPluginExists;
        if (options.plugins) {
            progressPluginExists = options.plugins.find((e) => e instanceof ProgressPlugin);
        }
```
I need help on this because i couldn't figure it out.
Thanks

**Comment by aman29271** [comment]
> After including `ProgressPlugin` in configuration file. Through debugging I found out `progerssPluginExists` is still undefined.
> 
> ```js
> const { ProgressPlugin } = webpack;
>         let progressPluginExists;
>         if (options.plugins) {
>             progressPluginExists = options.plugins.find((e) => e instanceof ProgressPlugin);
>         }
> ```
> 
> I need help on this because i couldn't figure it out.
> Thanks

I tried this instead...
```js
 progressPluginExists = options.plugins.find((e) => Object.getPrototypeOf(e).constructor.name === "ProgressPlugin");
```

**Comment by aman29271** [comment]
Closes #1497

**Comment by snitin315** [comment]
> Using --progress outputs all stages of compilation, percentage included. Previously, Output were overwritten by each other and only the final output got displayed. It also displays all stats information either provided through multiple entry or multiple config.

This is intentional, Refer https://github.com/webpack/webpack-cli/pull/1558#issuecomment-630618495

**Comment by evilebottnawi** [comment]
Fixed, feel free to feedback

**Pull request closed without merging**